test/api/search_controller_test.rb

```diff
@@ -0,0 +1,78 @@
+require 'api_test_helper'
+
```

Comment:
once you refactor more of the logic into static methods, this test class will  hopefully get simpler and faster :)